### PR TITLE
Fix new provider form and UI tests

### PR DESCRIPTION
**Pull request by mayorova**
**What this PR does / why we need it**:

Fixing some forms and tests that were apparently broken.

**Before:**
Screenshot from 2024-10-15 15-45-26

**After:**
Screenshot from 2024-10-15 15-51-23

As for the tests, I think this change has modified the behavior:
https://github.com/3scale/porta/pull/3919/files#diff-6784bb890636509ec8f3fec49d3b6d3af3b7d777beb990cd4b46e8721e1a99baR15-R17

Now for the fields that are marked required  it is not possible to submit empty values, because the browser prevents it (due to `required` property on the HTML).

So, I removed some tests that are not relevant, and updated other that can still be applied.

**Which issue(s) this PR fixes** 

**Verification steps** 

**Special notes for your reviewer**:

### CODE REVIEW

File: features/buyers/accounts/edit.feature [hunk 1]
```diff
@@ -19,12 +19,6 @@ Feature: Audience > Accounts > Edit
     Then the current page is the buyer account page for "Pepito"
     And should see the flash message "Account successfully updated"
 
-  Scenario: Required fields and validation
-    Given they go to the buyer account edit page for "Pepe"
-    When the form is submitted with:
-      | Organization/Group Name | |
-    Then field "Organization/Group Name" has inline error "can't be blank"
```
Comment (akostadinov) [inline features/buyers/accounts/edit.feature:26]: you mean that this test breaks with the HTML required?

Comment (mayorova) [reply features/buyers/accounts/edit.feature:26]: Yes, because you just cannot submit a form with the required field being empty.

Comment (akostadinov) [reply features/buyers/accounts/edit.feature:26]: Looks good. I am wondering though, can any of the answers here be used to validate that these fields do have the `required` property?

https://stackoverflow.com/questions/6114385

Or maybe we have some integration test where we can check the html of the form?

Comment (mayorova) [reply features/buyers/accounts/edit.feature:26]: > Or maybe we have some integration test where we can check the html of the form?

I don't think we have such a thing... but maybe I'm wrong.

Not sure what we need to add - whether these checks from SO (but I guess it is browser-dependant), or maybe unit tests for `output_html` method of the patternfly form builder.

But I'd vote for merging this PR as it is, and improve later :upside_down_face:

**Review by jlledom — APPROVED** [review]

File: features/buyers/accounts/new.feature [hunk 2]
```diff
@@ -30,12 +30,15 @@ Feature: Audience > Accounts > New
...
 
-  Scenario: Required fields and validation
+  Scenario: Fields validation
     Given they go to the new buyer account page
-    When press "Create"
+    When the form is submitted with:
+      | Username | u        |
+      | Email    | invalid  |
+      | Password | p        |
+      | Organization/Group Name | Org |
```
[the remark this answers is not in the document]
Comment (mayorova) [reply features/buyers/accounts/new.feature:39]: Yes, that's right. Except "presense" which is done through HTML `required` attribute.

**Pull request merged**